pkg/service/operation/submission/submission.go

```diff
@@ -0,0 +1,83 @@
+package submission
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/goccy/go-json"
+	"github.com/pkg/errors"
+	opstorage "github.com/tbd54566975/ssi-service/pkg/service/operation/storage"
+	"github.com/tbd54566975/ssi-service/pkg/storage"
+)
+
+// IDFromSubmissionID returns a submission operation ID from the submission ID.
+func IDFromSubmissionID(id string) string {
+	return fmt.Sprintf("%s/%s", ParentResource, id)
+}
+
+// ResourceID attempts to parse the submission id from the ID of the operation. This is done by taking the last word
+// that results from splitting the id by "/". On failures, the empty string is returned.
+func ResourceID(opID string) string {
+	i := strings.LastIndex(opID, "/")
+	if i == -1 {
+		return ""
+	}
+	return opID[(i + 1):]
+}
+
+// Namespace is the namespace to be used for storing submissions.
+const Namespace = "presentation_submission"
+
+// ParentResource is the prefix of the submission parent resource.
+const ParentResource = "/presentations/submissions"
+
+// Status indicates the current state of a submission.
+type Status uint8
```

Comment:
why a uint8 and not define a type for status

type (
	Status string
)

const (
	StatusPending  Status = "pending"
	StatusDenied Status = "denied"
)

Reply:
Synced offline. We're storing uints, and exposing strings.